feat: merge CAIP69 into CAIP10 for specifying EOA wallets (#107)
ChainAgnostic/namespaces: eip155/caip10.md

[CAIP-10][], there are still some particularities of the syntax that should be
specified for the unfamiliar.  

### Special case of EOA

> To express Ethereum account without reference to any specific [EIP-155][] chain ID, CAIP-10 identifiers can use the special `chainId` segment `0`.

Reviewer: needs link

Reviewer 2: oh sorry this is a weird elite markdown trick that we included in the style guide a while back-- that unfamiliar linking style actually gets its referent from the alias defined [here](https://github.com/ChainAgnostic/namespaces/blame/f521f2f1bbb71869af791becdd8017b3c70ab7d7/eip155/caip10.md#L103).  Most PR authors don't notice the instructions [here](https://github.com/ChainAgnostic/namespaces/blob/main/CONTRIBUTING.md#style-guide) in the `CONTRIBUTING.md` file so I just add a commit to everyone's PR translating the normal-human links to our kooky nerd link format.

> To express Ethereum account without reference to any specific [EIP-155][] chain ID, CAIP-10 identifiers can use the special `chainId` segment `0`.

Reviewer 3: CEOWALT

Note that a given address cannot be assumed to work on all current and future networks with an [EIP-155][] `chainId`, which may express or derive addresses differently for a given private key; a user account expressed with `chainId` segment of 0 is assumed to control the same account on `chainId` 1 and most ethereum chains, but MAY control a different account or additional accounts on networks that extend the account or addressing model of Ethereum, assume different key or hash derivations, etc.

## Syntax

Ethereum addresses were, historically, case-insensitive and normalized to use
all-lowercase letters (`abcdef`) like most hexadecimal numeric types.  With the
ratification of [EIP-55][], however, a particular normalization of lowercase- and
uppercase- `abcdefABCDEF` characters was invented as an efficient form of
checksum. See [EIP-55][] for specification.

The chain ID will be used to represent blockchain except special case of 0 as chainID to represent EOA.

The `chain_id` string will be ammended as follows:

```
chain_id:    namespace + ":" + network_id + ":" + reference
network_id:    [0-9]{1,19}
namespace:   [-a-z0-9]{3,8}

> reference:   0x[-_a-fA-F0-9]{1,32}

Reviewer 2: ```suggestion
reference:   0x[a-fA-F0-9]{1,32}
```
Whoops, missed this the first time.  Eth addresses can't have `-`s or `_`s in them! the CAIP-10 regex is much looser because it has to accomodate all kinds of addressing notations, but I think we can limit it to conventional EVM addresses here, at least until any L2s with custom addresses start complaining?

[CAIP-10][], there are still some particularities of the syntax that should be
specified for the unfamiliar.  

### Special case of EOA

> In order to enable decentralized applications to specify EOAs as an address, we propose using chainID 0. This would signify that it can live on multiple networks. In absence of this, addresses will be assigned to only one chain using CAIP-10 chainID syntax, an EOA would be required to be assigned multiple times for each chain. Specifying 0 as chainID for EOA, further enables EOAs to be designated as an EOA once, and be treated as an EOA on every applicable chain

Reviewer 2: Forgive me if these suggestions sound drastic, but I am thinking of some corner-cases and previous discussions with implementers which incline me to make subtle semantic adjustments.  Happy to discuss if this does not work with the intentions or use-cases you had in mind!

```suggestion
To express Ethereum account without reference to any specific [EIP-155][] chain ID, CAIP-10 identifiers can use the special `chainId` segment `0`.

In the Ethereum account system, "externally owned accounts" (i.e. "offchain" wallets) are controlled by user agents and not by on-chain entities (i.e. deployed or deployable smart contract code). 
The special chainId `0` SHOULD only be used for accounts that can be used in off-chain contexts, i.e. without the use of an Ethereum node.

Note that a given address cannot be assumed to work on all current and future networks with an [EIP-155][] `chainId`, which may express or derive addresses differently for a given private key; a user account expressed with `chainId` segment of 0 is assumed to control the same account on `chainId` 1 and most ethereum chains, but MAY control a different account or additional accounts on networks that extend the account or addressing model of Ethereum, assume different key or hash derivations, etc.
```

Reviewer 2: @ligi can you think of any additional information about the chainId registration system that should be added here or in a separate PR?  should 0 be "tombstoned" in eth-lists or was it already invalid?

Reviewer 3: K.I.S.S.RULE

uppercase- `abcdefABCDEF` characters was invented as an efficient form of
checksum. See [EIP55][] for specification.

The chain ID will be used to represent blockchain except special case of 0 as chainID to represent EOA.

The `chain_id` string will be ammended as follows:

```
chain_id:    namespace + ":" + network_id + ":" + reference

> network_id:    [0-9]{1,1}

Reviewer 2: ```suggestion
network_id:    [0-9]{1,19}
```
based on:
https://ethereum-magicians.org/t/eip-2294-explicit-bound-to-chain-id/11090/24
and
https://github.com/ethereum/EIPs/issues/2294
it seems chainId could be a bigint?

namespace:   [-a-z0-9]{3,8}

> reference:   [-_a-zA-Z0-9]{1,32}

Reviewer 2: ```suggestion
reference:   0x[-_a-fA-f-9]{1,32}
```

```

## Security Concerns

We should have users sign a message to prove they have an EOA (Greg to fill out more completely)

## References

> - [EIP155][]: Ethereum Improvement Proposal specifying generation and validation of ChainIDs

Reviewer 2: ```suggestion
- [EIP-155][]: Ethereum Improvement Proposal specifying generation and validation of ChainIDs
- [EIP-2294][]: Explicit Bound to ChainID Size
```

Reviewer 2: apologies for the style-guide chaos around EIPXXX versus EIP-XXX

Author: no worries. Thanks for the suggestion

```

> ## Security Concerns

Reviewer 2: ```suggestion
## Security Considerations
```

> We should have users sign a message to prove they have an EOA (Greg to fill out more completely)

Reviewer 2: ```suggestion
As the Ethereum namespace evolved, user-agents that connect to dapps through an [EIP-1193][] interface started to accrue "off-chain" use-cases such as authenticating control of an onchain account with an "off-chain" signature (i.e. a transaction signed by the wallet and verified by a dapp or website without either party submitting it to a node or running any on-chain functions). It is RECOMMENDED that wallets using the chainId of `0` be authenticated in a standardized and secure manner which produces verifiable receipts of the authenticaiton event, such as that documented in [ERC-4361][] and extended by [CAIP-122][].
```
LOL i also missed this the first time, too!  @gregthegreek did you have something additional in mind to add here, or is this enough?

Author: I will wait to commit this till Greg confirms

Reviewer: check my comment below!

## References

- [EIP-155][]: Ethereum Improvement Proposal specifying generation and validation of ChainIDs
- [ethereum-lists/chains][]: An open registry for eip155 network operators to claim a
      unique chainID and self-publish RPC/node information for them.
- [ERC-20][]: Basic [aka Fungible] Token Standard
- [ERC-721][]: Non-Fungible Token Standard

[Chainid.network]: https://github.com/ethereum-lists/chains
[CAIP-2]: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md
[CAIP-10]: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md
[CAIP-19]: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-19.md
[CAIP-21]: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-21.md
[CAIP-22]: https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-22.md

> [EIP-155]: https://eips.ethereum.org/EIPS/eip-155

Reviewer 2: ```suggestion
[EIP-1193]: https://eips.ethereum.org/EIPS/eip-1193
[EIP-155]: https://eips.ethereum.org/EIPS/eip-155
```
